Fix javadoc comments

[Javadoc specs have changed](http://www.oracle.com/technetwork/java/javase/documentation/index-137868.html#format) and compilation fails with the following issues with java 8

<code>
C:\Repos\Volley\volley\src\main\java\com\android\volley\toolbox\BasicNetwork.java:268: error: malformed HTML
     * Converts Headers[] to Map<String, String>.
                                ^
C:\Repos\Volley\volley\src\main\java\com\android\volley\toolbox\BasicNetwork.java:268: error: bad use of '>'
     * Converts Headers[] to Map<String, String>.
                                               ^
C:\Repos\Volley\volley\src\main\java\com\android\volley\VolleyLog.java:40: error: self-closing element not allowed
     * <br />
       ^
C:\Repos\Volley\volley\src\main\java\com\android\volley\VolleyLog.java:42: error: self-closing element not allowed
     * <br />
       ^
C:\Repos\Volley\volley\src\main\java\com\android\volley\VolleyLog.java:28: error: self-closing element not allowed
 * <p/>
   ^
C:\Repos\Volley\volley\src\main\java\com\android\volley\VolleyLog.java:29: error: self-closing element not allowed
 * to see Volley logs call:<br/>
                           ^
</code>

--- a/rules.gradle
+++ b/rules.gradle
@@ -20,11 +20,12 @@ if (configurations.findByName("testCompile")) {
   }
 }
 
-// TODO(#4): Remove this once Javadoc errors are fixed
-if (JavaVersion.current().isJava8Compatible()) {
-  allprojects {
-    tasks.withType(Javadoc) {
-      options.addStringOption('Xdoclint:none', '-quiet')
-    }
+android.libraryVariants.all { variant ->
+  task("generate${variant.name.capitalize()}Javadoc", type: Javadoc) {
+    source = variant.javaCompile.source
+    classpath = files(variant.javaCompile.classpath.files) +
+        files(project.android.getBootClasspath().join(File.pathSeparator))
+    options.linkSource true
+    failOnError true
   }
 }

--- a/rules.gradle
+++ b/rules.gradle
@@ -19,13 +19,3 @@ if (configurations.findByName("testCompile")) {
     testCompile "org.robolectric:robolectric:3.0"
   }
 }
-
-android.libraryVariants.all { variant ->
-  task("generate${variant.name.capitalize()}Javadoc", type: Javadoc) {
-    source = variant.javaCompile.source
-    classpath = files(variant.javaCompile.classpath.files) +
-        files(project.android.getBootClasspath().join(File.pathSeparator))
-    options.linkSource true
-    failOnError true
-  }
-}

--- a/src/main/java/com/android/volley/VolleyLog.java
+++ b/src/main/java/com/android/volley/VolleyLog.java
@@ -25,8 +25,8 @@ import java.util.Locale;
 
 /**
  * Logging helper class.
- * <p/>
- * to see Volley logs call:<br/>
+ * <p>
+ * to see Volley logs call:<br>
  * {@code <android-sdk>/platform-tools/adb shell setprop log.tag.Volley VERBOSE}
  */
 public class VolleyLog {
@@ -37,9 +37,9 @@ public class VolleyLog {
     /**
      * Customize the log tag for your application, so that other apps
      * using Volley don't mix their logs with yours.
-     * <br />
+     * <br>
      * Enable the log property for your tag before starting your app:
-     * <br />
+     * <br>
      * {@code adb shell setprop log.tag.&lt;tag&gt;}
      */
     public static void setTag(String tag) {

--- a/src/main/java/com/android/volley/toolbox/BasicNetwork.java
+++ b/src/main/java/com/android/volley/toolbox/BasicNetwork.java
@@ -265,7 +265,7 @@ public class BasicNetwork implements Network {
     }
 
     /**
-     * Converts Headers[] to Map<String, String>.
+     * Converts Headers[] to Map&lt;String, String&gt;.
      */
     protected static Map<String, String> convertHeaders(Header[] headers) {
         Map<String, String> result = new TreeMap<String, String>(String.CASE_INSENSITIVE_ORDER);

--- a/src/main/java/com/android/volley/toolbox/HttpHeaderParser.java
+++ b/src/main/java/com/android/volley/toolbox/HttpHeaderParser.java
@@ -31,7 +31,7 @@ import java.util.Map;
 public class HttpHeaderParser {
 
     /**
-     * Extracts a {@link Cache.Entry} from a {@link NetworkResponse}.
+     * Extracts a {@link com.android.volley.Cache.Entry} from a {@link NetworkResponse}.
      *
      * @param response The network response to parse headers from
      * @return a cache entry for the given response, or null if the response is not cacheable.

